Allow initialization of multiple devices

pvint opened the pull request:
Re #21 

When attempting to initialize more than one device *all* devices on the bus were being reset when PCA9685_initPWM() was called. Perhaps there was a reason to do so, but it looks to me like it's fine to simply reset the desired device.

This change resets only the *one* device, and I have tested with two devices, one at 0x40 and one at 0x60 and all looks good in my testing.

I also added changes to examples/quickstart/quickstart.c to demonstrate this. (I added an option for a second device in seizure mode, but if you set the second address to 0x00 it ignores it)

Thanks to @deelerke for contacting me and giving me a bump on this!

pvint committed:
leave /dev/i2c-1 as 0660

pvint committed:
Changed "reset all" in initPWM() to just reset the selected device (also modified example to demonstrate second device usage)

pvint committed:
Merge branch 'develop' of https://github.com/pvint/libPCA9685 into develop

pvint committed:
Added comment info

pvint committed:
Fixed info in comment

deelerke commented:
thanks @pvint, I tested this with three devices on address 40, 41 and 42 and they all work fine ! very nice

edlins commented:
This looks okay.  I love one line fixes.  If quickstart was part of the core library, I would've written those changes differently, but it's fine for quickstart.  I'll see if I can fix the Travis CI build..

edlins commented:
Okay, when `PCA9685_initPWM()` was changed from
```
ret = _PCA9685_writeI2CRaw(fd, _PCA9685_GENCALLADDR, 1, &resetval);
```
to
```
ret = _PCA9685_writeI2CRaw(fd, addr, 1, &resetval);
```
The `addr` passed in to `_PCA9685_writeI2CRaw()` was changed from `_PCA9685_GENCALL` (value `0x00`) to the actual `addr` (value `0x40`).  Consequently, the actual output of `PCA9685test` logged the actual `addr` value `0x40` instead of the original expected `GENCALL` value `0x00`.
So the `PCA9685_expected_output` file needs to be updated or regenerated after the lib change.  The file in the `test` folder can be edited to match (not great) or regenerated by running
```
./test/PCA9685test -td 1 40 > PCA9685_actual_output
```
as executed by `CMakeLists.txt` and logged in the Travis CI build.  If regenerated with the above command, rename `PCA9685_actual_output` to `PCA9685_expected_output` and move it in the `test` folder to replace the old expected output.
After the new expected output file is added, the Travis CI build should pass.

edlins commented on src/PCA9685.c:
```diff
@@ -70,7 +70,7 @@ int PCA9685_initPWM(int fd, unsigned char addr, unsigned int freq) {
 
   // send a software reset to get defaults, resets all devices on bus
```
Please update this comment to reflect the change

edlins replied:
Drat, one more thing..  The `README.md` will also need to be updated in `PCA9685_initPWM` to state that only the device at `addr` is reset (instead of "all-devices reset").

pvint replied:
Ooops, I had already fixed that comment locally but hadn't pushed it ;)

edlins commented:
In summary (sorry for the mess of comments):

1. Fix comment in `PCA9685_initPWM()` in `PCA9685.c`
2. Fix `addr` values in `PCA9685_expected_output`
3. Fix `PCA9685_initPWM` documentation in `README.md`

pvint commented:
I'll try to do this this evening - Thanks Scott!

pvint committed:
Updated info and fixed Travis CI test

pvint commented:
Did the updates - the Travis build is failing with something related to olaclient now. 

That's all I have time for tonight - I can have a look at that tomorrow night.

Paul

edlins committed:
capture log

Travis CI build fails bc unit file not capturing stdout and stderr to /var/log/

edlins committed:
revert, didn'

edlins committed:
Update .travis.yml

comment out olaclient log check.  olaclient seems to fail in Travis env..

pvint commented:
No problem, thank you!

Just as an aside: I used this heavily in controlling all of my lighting on my boat all of last year, and the only problems I had were on a higher level (ie: my code that was using this library). IMO, it's been proven to be stable and reliable, and maybe time for a 1.0. :)
